reeprotocol/app_asdu.py

```diff
@@ -184,33 +188,182 @@ def length(self):
         return 0x15
 
 
-class M_IT_TK_2(BaseAppAsdu):
+class C_CI_NT_2(C_CI_NX_2):
     """
-    Totales integrados operacionales repuestos periódicamente, 4 octetos
-    (incrementos de energía, en kWh o kVARh)
+    Class to read absolute measures
     """
+    type = 122
 
-    type = 11
+
+class C_CI_NU_2(C_CI_NX_2):
+    """
+    Class to read incremental measures
+    """
+    type = 123
+
+
+class C_TA_VC_2(BaseAppAsdu):
+    """
+    Leer Información de Tarificación (Valores en Curso)
+    """
+    type = 133
+    causa_tm = 6
+
+    # Register address: 134 or 135 or 136
+
+    def from_hex(self, data, cualificador_ev):
+        pass
+
+    def to_bytes(self):
+        return bytes()
+
+
+class C_TA_VM_2(BaseAppAsdu):
+    """
+    Leer Información de Tarificación (Valores Memorizados)
+    """
+    type = 134
+    data_length = 0x06
+    causa_tm = 6
+
+    # Register address: 134 or 135 or 136
+    def __init__(self, start_date=datetime.datetime.now(),
+                 end_date=datetime.datetime.now()):
+        self.start_date = TimeA(start_date)
+        self.end_date = TimeA(end_date)
+
+    def from_hex(self, data, cualificador_ev):
+        self.start_date.from_hex(data[0:5])
+        self.end_date.from_hex(data[5:10])
+
+    def to_bytes(self):
+        response = bytearray()
+        response.extend(self.start_date.to_bytes())
+        response.extend(self.end_date.to_bytes())
+        return response
+
+    @property
+    def length(self):
+        return 0x13
+
+
+class M_TA_VX_2(BaseAppAsdu):
+    """
+    Class for the M_TA_VC_2(135) and M_TA_VM_2(136) ASDUs
+    """
+    data_length = 0x06
+    causa_tm = 5
+    BillingRegister = namedtuple('BillingRegister', ['address', 'active_abs',
+    'active_inc', 'active_qual', 'reactive_abs_ind', 'reactive_inc_ind',
+    'reactive_qua_ind', 'reactive_abs_cap', 'reactive_inc_cap',
+    'reactive_qual_cap', 'reserved_7', 'reserved_7_qual', 'reserved_8',
+    'reserved_8_qual', 'max_power', 'max_power_date', 'max_power_qual',
+    'excess_power', 'ecxess_power_qual', 'date_start', 'date_end'])
```

Comment:
Define this type outside of the `M_TA_VX_2` class

Reply:
@ecarreras Moved the definition of the type and added the required tests.